### PR TITLE
Doc comments: fix validation issues in docs build

### PR DESCRIPTION
**Pull request by ghogen**
Fixes https://github.com/dotnet/msbuild/issues/6743

### Context
The doc build is throwing some validation errors on some XML code comment constructs, such as invalid tags or improper nesting of tags.

### Changes Made
Fix several XML comments

### Notes
No change other than code comments

### CODE REVIEW

**Comment by ghogen** [comment]
@gewarren Here are the changes in MSBuild for the doc validation issues.

File: src/Tasks/CreateItem.cs [hunk 1]
```diff
@@ -31,14 +31,12 @@ public class CreateItem : TaskExtension
...
-        /// The fact that this is a string[] makes the following illegal:
-        ///     <CreateItem
-        ///         AdditionalMetadata="TargetPath=@(OutputPathItem)" />
-        /// The engine fails on this because it doesn't like item lists being concatenated with string
-        /// constants when the data is being passed into an array parameter.  So the workaround is to 
-        /// write this in the project file:
-        ///     <CreateItem
-        ///         AdditionalMetadata="@(OutputPathItem->'TargetPath=%(Identity)')" />
+        ///    The fact that this is a `string[]` makes the following illegal:
+        ///      `<CreateItem AdditionalMetadata="TargetPath=@(OutputPathItem)" />`
```
Comment (gewarren) [inline src/Tasks/CreateItem.cs:35]: You can't use backticks for code blocks unless you put this in a CDATA section of Markdown.

File: src/Tasks/MSBuild.cs [hunk 2]
```diff
@@ -55,14 +55,12 @@ private enum SkipNonexistentProjectsBehavior
...
-        /// The fact that this is a string[] makes the following illegal:
-        ///     <MSBuild
-        ///         Properties="TargetPath=@(OutputPathItem)" />
-        /// The engine fails on this because it doesn't like item lists being concatenated with string
-        /// constants when the data is being passed into an array parameter.  So the workaround is to 
-        /// write this in the project file:
-        ///     <MSBuild
-        ///         Properties="@(OutputPathItem->'TargetPath=%(Identity)')" />
+        ///     The fact that this is a `string[]` makes the following illegal:
+        ///       `<MSBuild Properties="TargetPath=@(OutputPathItem)" />`
```
Comment (gewarren) [inline src/Tasks/MSBuild.cs:59]: Same here.

File: src/Build/Definition/Project.cs [hunk 3]
```diff
@@ -88,7 +88,7 @@ public class Project : ILinkableObject
         /// - <see cref="ProjectItem.SetMetadataValue(string,string)"/>
         /// - <see cref="ProjectItem.SetMetadataValue(string,string, bool)"/>
         ///
-        /// When this property is set to true, the previous item operations throw an <exception cref="InvalidOperationException"></exception>
+        /// When this property is set to true, the previous item operations throw an <see cref="InvalidOperationException"></see>
```
Comment (rainersigwald) [inline src/Build/Definition/Project.cs:91]:
```suggestion
        /// When this property is set to true, the previous item operations throw an <see cref="InvalidOperationException" />
```

?

**Review by rainersigwald — APPROVED** [review]

**Comment by ghogen** [comment]
@rainersigwald  porting doc comment changes from this PR into MSBuild source:
https://github.com/dotnet/dotnet-api-docs/pull/7082

**Pull request merged**